src/os/unix/unix.cpp

```diff
@@ -213,12 +213,12 @@ const char *FS2OTTD(const char *name)
 		}
 	}
 
-	return convert_tofrom_fs(convd, name);
+	return convert_tofrom_fs(convd, name.c_str(), buf, lengthof(buf));
 }
 
 #else
-const char *FS2OTTD(const char *name) {return name;}
-const char *OTTD2FS(const char *name) {return name;}
+std::string FS2OTTD(const std::string &name) {return name;}
+std::string OTTD2FS(const std::string &name) {return name;}
```

Comment:
This seems a bit of a pessimisation for non-Windows platforms.
This will always create a copy of the input, even if the input could otherwise be an rvalue which could be moved from, or if the result is passed straight through to c_str(), in which no copy is needed at all.

Reply:
I guess maybe defining them as a template function?
```c++
template <typename T> std::string FS2OTTD(T name) { return name; }
```
I think that will avoid all unnecessary copies?

Reply:
https://godbolt.org/z/7hdd1b Suggests that the templated version generates the simplest code

Reply:
Oh, turns out the templated version is not quite possible, since `stdafx.h` has just a forward declaration, and the implementations are in `unix.cpp` (null-implementation and iconv implementation) and `os2.cpp` (null-implementation). I guess `stdafx.h` could select on `defined(WITH_ICONV)` whether to forward declare or use the templated version.